jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java

```diff
@@ -0,0 +1,153 @@
+/*[INCLUDE-IF Sidecar18-SE]*/
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package openj9.tools.attach.diagnostics.base;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import com.ibm.tools.attach.target.IPC;
+
+/**
+ * Common methods for the diagnostics tools
+ *
+ */
+public class DiagnosticUtils {
+
+	/**
+	 * Command strings for executeDiagnosticCommand()
+	 */
+	
+	/**
+	 * Run System.gc();
+	 */
+	public static final String DIAGNOSTICS_GC_RUN = "GC.run"; //$NON-NLS-1$
+
+	/**
+	 * Get the heap object statistics.
+	 */
+	public static final String DIAGNOSTICS_GC_CLASS_HISTOGRAM = "GC.class_histogram"; //$NON-NLS-1$
+
+	/**
+	 * Key for the command sent to executeDiagnosticCommand()
+	 */
+	public static final String COMMAND_STRING = "command_string"; //$NON-NLS-1$
+	
+	/**
+	 * Use this to separate arguments in a diagnostic command string.
+	 */
+	public static final String DIAGNOSTICS_OPTION_SEPARATOR = ","; //$NON-NLS-1$
+
+	/**
+	 * Report only live heap objects.
+	 */
+	public static final String LIVE_OPTION = "live"; //$NON-NLS-1$
+
+	static final Map<String, Function<String, DiagnosticProperties>> commandTable;
+	/**
+	 * @return contents of stdin as a list of strings, one per line.
+	 */
+	public static List<String> stdinToStringList() {
```

Comment:
This shouldn't be in java.base

Reply:
Moved to a new utility class. Refactored the same code out of `jstack`.